### DISCUSSION TITLE
How can I get the usage with `GetStreamingChatMessageContentsAsync`

### DISCUSSION THREAD

**Discussion by SebastianStehle** [microsoft/semantic-kernel#5624] [Q&A] [answered]
Hi,

for the non-streaming method `GetChatMessageContentsAsync` I get the usage like this:

```
private decimal CalculateCosts(IReadOnlyList<ChatMessageContent> result)
{
    var costs = 0m;

    if (result[0].Metadata?.TryGetValue("Usage", out var m) == true && m is CompletionsUsage usage)
    {
        costs += usage.PromptTokens * options.PricePerInputTokenInEUR;
        costs += usage.CompletionTokens * options.PricePerOutputTokenInEUR;
    }

    return costs;
}
```

This works great. But for the streaming method, none of the items have a usage metadata right now. I am using OpenAI.

**Comment by moonbox3** [comment, marked as answer]
Hi @SebastianStehle, this would be served as an issue instead of a question in Q&A. Could you please file this issue so we can track it and investigate? Thank you. Let's then close this question.

**Reply by SebastianStehle** [reply]
Done. I thought I have just missed it somehow.
